api/src/service/RpcClient.ts

```diff
@@ -162,4 +178,56 @@ export class RpcClient {
         });
     });
   }
+
+  private encryptItems = (params: any[]) => {
+    let encryptedParams: any[];
+    encryptedParams = params.map((obj, index) => {
+      let encryptedObjectAsPlainString = "";
+      if (index === 0 || index === 1 || obj === "offchain") {
+        // streamname, keys, offchain-flag are not encrypted
+        return obj;
+      }
+      if (config.encryptionKey) {
+        encryptedObjectAsPlainString = encrypt(config.encryptionKey, JSON.stringify(obj));
+      }
+      // Wrapping encrypted plaintext into object for multichain
+      return { json: encryptedObjectAsPlainString };
+    });
+    params = encryptedParams;
+    return params;
+  };
+
+  private decryptItems = (responseData: any) => {
+    return responseData?.map((obj) => {
+      let decryptedDataResult = {};
+      if (obj.data.json) {
+        const plainStringToDecrypt = obj.data.json;
+        if (config.encryptionKey) {
+          if (typeof plainStringToDecrypt !== "string") {
+            throw new Error(
+              "failed to decrypt events: expected string, but got other type instead. This error usually occurs when you set the ENCRYPTION_KEY at an unencrypted blockchain. To resolve this error, unset the env var ENCRYPTION_KEY.",
+            );
+          }
+          const jsonString = decrypt(config.encryptionKey, plainStringToDecrypt);
+          if (Result.isErr(jsonString)) {
+            throw new Error("failed to decrypt events from blockchain. Is ENCRYPTION_KEY set correctly?");
+          }
```

Comment:
Here, we can't use VError because this is not a route. Instead, the error has to be thrown.

![image](https://user-images.githubusercontent.com/59825636/127486408-a5ddc63c-173b-46ec-9ad7-810b63c7933d.png)